spec/services/expense_services_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe ExpenseService, type: :service do
+
+  let(:group) { create(:group) }
+  let(:split_type) { create(:split_type) }
+  let(:user1) { create(:user) }
+  let(:user2) { create(:user) }
+  let(:expense_params) { { total_amount: 200, description: 'Test expense', split_type_id: split_type.id, date: Date.today } }
+  let(:invalid_expense_params) { { total_amount: nil, description: 'Test expense', split_type_id: split_type.id, date: Date.today } }
+  let(:payment_details) { [{ user_id: user1.id, paid_amount: 100 }, { user_id: user2.id, paid_amount: 200 }] }
+  let(:service) { described_class.new(group, expense_params, payment_details) }
+  let(:invalid_service) { described_class.new(group, invalid_expense_params, payment_details) }
+
+  describe '#create_expense' do
+    context 'when expense is successfully created' do
+      it 'returns the created expense' do
+        result = service.create_expense
+        expect(result[:expense]).to be_persisted
+        expect(result[:error]).to be_nil
+      end
+    end
+
+    context 'when expense fails to save' do
+      before do
+        allow_any_instance_of(Expense).to receive(:save).and_return(false)
+      end
+
+      it 'returns an error message' do
+        result = invalid_service.create_expense
+        expect(result[:expense]).to be_nil
+        expect(result[:error]).not_to be_nil
+      end
+    end
+
+    context 'when a CustomError is raised' do
+      before { allow(service).to receive(:process_payment_details).and_raise(CustomError, 'Custom error message') }
+
+      it 'destroys the expense and returns the error message' do
+        result = service.create_expense
+        expect(result[:expense]).to be_nil
+        expect(result[:error]).to eq('Custom error message')
+      end
+    end
+  end
+
+  describe '#process_payment_details' do
```

Comment:
No need of testing this method independently. Since this method is not called from anywhere else (other than from the same service class), it should ideally be marked as a `private` method. 

We don't generally write unit tests for private methods. Instead we add test scenarios in the unit tests of the methods that calls these private methods and ensure all the cases are covered.